Upgrade from LaunchConfig -> LaunchTemplate causes Segfault

**Is this a BUG REPORT or FEATURE REQUEST?**:
Bug 
**What happened**:
Controller crashes when type is changing from LaunchConfig to LaunchTemplate
**What you expected to happen**:
It should not crash and should change types correctly. 
**How to reproduce it (as minimally and precisely as possible)**:
Change an IG from LaunchConfig to LaunchTemplate
**Anything else we need to know?**:
Investigating - due to LaunchTemplate version changes. 
**Environment**:

- Kubernetes version:

```bash
kubectl version -o yaml
```

**Other debugging information (if applicable)**:

- InstanceGroup status:

```bash
kubectl describe instancegroup <ig-name>
```

- controller logs:

```
E0212 00:01:51.691425       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 338 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1cf5fa0, 0x3062fb0)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.6/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.6/pkg/util/runtime/runtime.go:48 +0x89
panic(0x1cf5fa0, 0x3062fb0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/keikoproj/instance-manager/controllers/provisioners/eks.(*EksInstanceGroupContext).UpdateScalingGroup(0xc000563810, 0xc0010c4ba0, 0x2c, 0xc00102b538, 0x4, 0x4)
	/workspace/controllers/provisioners/eks/update.go:167 +0x38d
github.com/keikoproj/instance-manager/controllers/provisioners/eks.(*EksInstanceGroupContext).Update(0xc000563810, 0x1fd8690, 0xa)
	/workspace/controllers/provisioners/eks/update.go:103 +0x7bf
github.com/keikoproj/instance-manager/controllers.HandleReconcileRequest(0x22ff960, 0xc000563810, 0x0, 0xc0004ec5a0)
	/workspace/controllers/interface.go:49 +0x3d1
github.com/keikoproj/instance-manager/controllers.(*InstanceGroupReconciler).Reconcile(0xc0002d2770, 0x22ea980, 0xc001062b70, 0xc000765dc9, 0x5, 0xc000765dc4, 0x5, 0xc001062b70, 0x40a1ff, 0xc000030000, ...)
	/workspace/controllers/instancegroup_controller.go:188 +0x879
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00071b0e0, 0x22ea8c0, 0xc00076a2c0, 0x1d99180, 0xc0003fc560)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:263 +0x317
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00071b0e0, 0x22ea8c0, 0xc00076a2c0, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:235 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.1(0x22ea8c0, 0xc00076a2c0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:198 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	/go/pkg/mod/k8s.io/apimachinery@v0.19.6/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000278750)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.6/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00102bf50, 0x22a9660, 0xc001062ae0, 0xc00076a201, 0xc0006c27e0)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.6/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000278750, 0x3b9aca00, 0x0, 0x1, 0xc0006c27e0)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.6/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x22ea8c0, 0xc00076a2c0, 0xc00091bcd0, 0x3b9aca00, 0x0, 0xc000047401)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.6/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x22ea8c0, 0xc00076a2c0, 0xc00091bcd0, 0x3b9aca00)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.6/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:195 +0x4e7
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereferenc
```

--- a/controllers/provisioners/eks/update.go
+++ b/controllers/provisioners/eks/update.go
@@ -164,11 +164,15 @@ func (ctx *EksInstanceGroupContext) UpdateScalingGroup(configName string, scalin
 				Version:            aws.String("$Latest"),
 			}
 		}
-		latestVersion := (*scalingConfig).(*scaling.LaunchTemplate).LatestVersion.VersionNumber
-		latestVersionString := strconv.FormatInt(*latestVersion, 10)
 
 		status.SetActiveLaunchTemplateName(configName)
-		status.SetLatestTemplateVersion(latestVersionString)
+		switch scalingConfigType := (*scalingConfig).(type) {
+		case *scaling.LaunchTemplate:
+			latestVersion := scalingConfigType.LatestVersion.VersionNumber
+			latestVersionString := strconv.FormatInt(*latestVersion, 10)
+			status.SetLatestTemplateVersion(latestVersionString)
+		}
+
 	}
 
 	if ctx.ScalingGroupUpdateNeeded(configName) {

--- a/controllers/provisioners/eks/update.go
+++ b/controllers/provisioners/eks/update.go
@@ -171,6 +171,9 @@ func (ctx *EksInstanceGroupContext) UpdateScalingGroup(configName string, scalin
 			latestVersion := scalingConfigType.LatestVersion.VersionNumber
 			latestVersionString := strconv.FormatInt(*latestVersion, 10)
 			status.SetLatestTemplateVersion(latestVersionString)
+		case *scaling.LaunchConfiguration:
+			//LaunchConfiguration to LaunchTemplate migration. Latest version is the initial version.
+			status.SetLatestTemplateVersion("1")
 		}
 
 	}

